VRDR/DeathRecord_submissionProperties.cs

```diff
@@ -353,11 +362,15 @@ public string StateSpecific
                         return stateSpecificData.Value.ToString();
                     }
                 }
-                return "";
+                return null;
             }
             set
             {
                 // TODO: Handle case where Composition == null (either create it or throw exception)
+                if (String.IsNullOrWhiteSpace(value))
+                {
+                    return;
+                }
                 Composition.Extension.RemoveAll(ext => ext.Url == ExtensionURL.StateSpecificField);
```

Comment:
From Saul's email, fields like State Specific that have a RemoveAll or some sort of delete that clears the field should do the delete first and then check whether the value is null and create the extension if its not null or white space.